### PR TITLE
Send invoice with own mail framework

### PR DESCRIPTION
**Pull request by Ninjeneer**
image

### CODE REVIEW

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
Provide template for ES and DE.

File: src/integration/billing/BillingIntegration.ts [hunk 1]
```diff
@@ -364,6 +366,28 @@ export default abstract class BillingIntegration<T extends BillingSetting> {
         detailedMessages: { error: error.message, stack: error.stack }
       });
     }
+    const tenant = await TenantStorage.getTenant(this.tenantID);
```
Comment (LucasBrazi06) [inline src/integration/billing/BillingIntegration.ts:369]: I would expect the code above (line 351 -> 368 to not be in that method but in the one which create the invoice.
Here I want only to see the code to send the invoice that's it.

File: src/integration/billing/BillingIntegration.ts [hunk 2]
```diff
@@ -364,6 +366,28 @@ export default abstract class BillingIntegration<T extends BillingSetting> {
...
+      Utils.generateGUID(),
+      invoice.user,
+      {
+        user: invoice.user,
+        evseDashboardInvoiceURL: await Utils.buildEvseBillingInvoicesURL(this.tenantID),
+        evseDashboardURL: Utils.buildEvseURL(tenant.subdomain),
+        invoiceDownloadUrl: await Utils.buildEvseBillingDownloadInvoicesURL(this.tenantID, invoice.id),
+        invoice: invoice
+      }
+    ).catch((error) => {
```
Comment (LucasBrazi06) [inline src/integration/billing/BillingIntegration.ts:382]: Don't use this form of catch here, this is an async call and we don't want to handle errors here.

File: src/integration/billing/BillingIntegration.ts [hunk 3]
```diff
@@ -364,6 +366,28 @@ export default abstract class BillingIntegration<T extends BillingSetting> {
         detailedMessages: { error: error.message, stack: error.stack }
       });
     }
+    const tenant = await TenantStorage.getTenant(this.tenantID);
+    // Send async notification
+    NotificationHandler.sendBillingNewInvoiceNotification(
+      this.tenantID,
+      Utils.generateGUID(),
```
Comment (LucasBrazi06) [inline src/integration/billing/BillingIntegration.ts:373]: You can pass the invoice ID here

File: src/utils/Utils.ts [hunk 4]
```diff
@@ -1055,28 +1055,33 @@ export default class Utils {
...
+  }
+
+  public static async buildEvseBillingInvoicesURL(tenantID: string): Promise<string> {
+    const tenant = await TenantStorage.getTenant(tenantID);
+    return `${Utils.buildEvseURL(tenant.subdomain)}/invoices`;
+  }
+
+  public static async buildEvseBillingDownloadInvoicesURL(tenantID: string, invoiceID: string): Promise<string> {
+    const tenant = await TenantStorage.getTenant(tenantID);
+    return `${Utils.buildEvseURL(tenant.subdomain)}/invoices?InvoiceID=${invoiceID}#all`;
```
Comment (LucasBrazi06) [inline src/utils/Utils.ts:1084]: I removed all the useless intermediate constants in all buildxxxURL() methods.
I changed the invoiceID -> InvoiceID (first letter in uppercase when using Query params)
Update the code in the front-end to handle this change

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
- Send translation files to Tobias for Portuguese ongoing translation and integrate it back in this PR.
- Change the link to a button:
image

File: src/integration/billing/BillingIntegration.ts (hunk 1 above)
Comment (Ninjeneer) [reply src/integration/billing/BillingIntegration.ts:352]: Moved it into the concrete billing implementation, as the invoice finalization depends on how the billing provider works.
In this case with Stripe, not at the invoice creation but at the state switch DRAFT -> OPEN

File: src/integration/billing/BillingIntegration.ts (hunk 2 above)
Comment (Ninjeneer) [reply src/integration/billing/BillingIntegration.ts]: Removed

File: src/integration/billing/BillingIntegration.ts (hunk 3 above)
Comment (Ninjeneer) [reply src/integration/billing/BillingIntegration.ts]: Done

**Comment by Ninjeneer** [comment]
I had to change the body template structure, to handle multiple buttons in mail notifications, which explains the hundreds of edited files. Nothing relevant, just a massive addition of brackets.

**Pull request merged**